lib/ecto/repo/schema.ex

```diff
@@ -159,6 +171,10 @@ defmodule Ecto.Repo.Schema do
 
   defp init_mapper(schema, dumper, adapter, placeholder_map) do
     fn {field, value}, acc ->
+      unless writable_field?(schema, field) do
+        raise ArgumentError, "cannot give unwritable field `#{field}` to insert_all"
+      end
+
       case dumper do
         %{^field => {source, type}} ->
```

Comment:
Should we instead not include read only fields in the dumper?

Reply:
Oh we need them in the dumper because of `select_dump` in the planner.

Reply:
We could make a `write_dumper` and use that here instead?

Reply:
> Either that or I could add read_only? to the current dumper output

Good call! Let's try that, given the purpose of the dumper was to encapsulate all info in a single place.